bindings/cpp/posix_utils.cc

```diff
@@ -0,0 +1,53 @@
+#include <wtf/runtime.h>
+#include <signal.h>
+
+#include <thread>
+#include <condition_variable>
+
+namespace {
+
+std::condition_variable gCv;
+std::mutex gMutex;
+std::string gFilename;
+
+void wtfOnSignal(int) {
+  gCv.notify_all();
+};
+
+void wtfSignalWatcherThread () {
+  std::unique_lock<std::mutex> lock { gMutex };
+  while (true) {
+    gCv.wait(lock);
+    wtf::Runtime::GetInstance()->SaveToFile(gFilename.c_str());
+  }
+}
+
+} // namespace
+
+namespace wtf {
+namespace posix_utils {
+
+void InstallSignalHandler(int number, char const* filename) {
+  if (!filename) {
+    fprintf(stderr, "%s: Error: filename can't be null\n", __PRETTY_FUNCTION__);
+    return;
+  }
+
+  static bool onlyOnce = false;
```

Comment:
This is still a bit racy. I typically see a std::atomic<bool> used for this (with a condition on if (only_once.exchange(true))).